docs/standard/serialization/system-text-json/snippets/custom-contracts/IgnoreType.cs

```diff
@@ -0,0 +1,67 @@
+using System.Text.Json;
+using System.Text.Json.Serialization.Metadata;
+
+namespace Serialization
+{
+    class ExampleClass
+    {
+        public string Name { get; set; } = "";
+        public SecretHolder? Secret { get; set; }
+    }
+
+    class SecretHolder
+    {
+        public string Value { get; set; } = "";
+    }
+
+    class IgnorePropertiesWithType
+    {
+        private List<Type> _ignoredTypes = new List<Type>();
+
+        public void IgnorePropertyWithType(Type type)
+        {
+            _ignoredTypes.Add(type);
+        }
+
+        public void ModifyTypeInfo(JsonTypeInfo ti)
+        {
+            if (ti.Kind != JsonTypeInfoKind.Object)
+                return;
+
+            JsonPropertyInfo[] props = ti.Properties.Where((pi) => !_ignoredTypes.Contains(pi.PropertyType)).ToArray();
```

Comment:
Filtering properties is awkward without an `IList<T>.RemoveAll` method, but it can be easily implemented as an extension method:

```C#
class IgnorePropertiesWithType
{
    private readonly Type[] _ignoredTypes;

    public IgnorePropertiesWithType(params Type[] ignoredTypes)
        => _ignoredTypes = ignoredTypes;

    public void ModifyTypeInfo(JsonTypeInfo ti)
    {
        if (ti.Kind != JsonTypeInfoKind.Object)
            return;

        ti.Properties.RemoveAll(prop => _ignoredTypes.Contains(prop.PropertyType));
    }
}

public static class ListHelpers
{
    // IList<T> implementation of List<T>.RemoveAll method
    public static void RemoveAll<T>(this IList<T> list, Predicate<T> predicate)
    {
        for (int i = 0; i < list.Count; i++)
        {
            if (predicate(list[i]))
            {
                list.RemoveAt(i--);
            }
        }
    }
}
```

Reply:
API proposal created: https://github.com/dotnet/runtime/issues/76375